force-app/main/default/classes/DMLOperations.cls

```diff
@@ -20,199 +20,345 @@
+   * The Opportunity's Stage is set to Qualification.
+   * The Opportunity's Close Date is set to 3 months from today.
+   * The Opportunity's Amount is set to 50000.
+   * @param list of Opportunities to be changed then upserted.
+   */
+  public static void upsertOpportunityList(List<Opportunity> opportunities) {
+    // Loop through each Opportunity in the list
+    // Set the Opportunity's Stage to Qualification
+    // Set the Opportunity's Close Date to 3 months from today
+    // Set the Opportunity's Amount to 50000
+    for (Opportunity opp : opportunities) {
+      opp.StageName = 'Qualification';
+      opp.CloseDate = System.today().addMonths(3);
+      opp.Amount = 50000;
     }
-
-    /**
-     * Question 2
-     * Create a new Account with the specified name and industry provided.
-     * @param name The name of the account to be created.
-     * @param industry The industry of the account to be created.
-     */
-    public static void createAccount(String name, String industry) {
-        // Create a new Account and set the fields inline
-
-        // Insert the Account
+    // Upsert the opportunities List so that records are either updated or created
+    upsert as user opportunities;
+  }
+
+  /**
+   * Question 8
+   * Insert or updates opportunities from list of names.
+   * Associate the Opportunities with account that has the name accountName.
+   * There should be an Opportunity for each name provided in the list.
+   * @param accountName The name of the Account.
+   * @param oppNames The list of Opportunity names.
+   */
+  public static void upsertOpportunities(
+    String accountName,
+    List<String> oppNames
+  ) {
+    List<Opportunity> oppsToUpsert = new List<Opportunity>();
+    Set<Opportunity> oppsToUpsertSet = new Set<Opportunity>();
+    List<Account> acctLst = [
+      SELECT Id
+      FROM Account
+      WHERE Name = :accountName
+      LIMIT 1
+    ];
+    Account acct;
+    if (acctLst.isEmpty()) {
+      acct = new Account(name = accountName);
+      insert as user acct;
+    } else {
+      acct = acctLst[0];
     }
-
-    
-    /**
-     * Question 3
-     * Create a new Contact and link it to the Account id provided then inserts it into the database.
-     * @param accountId The Id of the Account to link the Contact to.
-     * @return The Id of the newly created Contact.
-     */
-    public static Id insertNewContact(Id accountId) {
-        // Create a new Contact instance
-
-        // Set some fields on the Contact
-        // Set the AccountId field to the given accountId
-
-        // Insert the Contact into the database
-
-        // Return the Id of the newly created Contact
-        return null;
+    List<Opportunity> existingOpps = [
+      SELECT Id, Name, AccountId
+      FROM Opportunity
+      WHERE Name IN :oppNames AND AccountId = :acct.Id
+    ];
+    // Loop through each Opportunity name
+    // Add the Opportunity to the list
+    for (Integer i = 0; i < oppNames.size(); i++) {
```

Comment:
Rather than having nested for loops, adding the list of existing Opportunities to a Map would enable you to loop over the list of Opportunity names and then check if the Map contains the name for that iteration in its key.